# Regression Information...
- Which feature did you attempt to predict? What was the reported prediction score? Is this feature is necessary for identifying customers' spending habits?
- Hint: The coefficient of determination, R^2, is scored between 0 and 1, with 1 being a perfect fit. A negative R^2 implies the model fails to fit the data.
- - With a coefficient of determination of .681, we know the Grocery feature does describe some of the same information found in the other features. So .319 is not enough "unique feature information" to consider it a necessary feature.

In [5]:
import pandas as pd

df = pd.read_csv("sysco.csv")

df.head()

Fresh  Milk  Grocery  Frozen  Detergents_Paper  Delicatessen
0  12669  9656     7561     214              2674          1338
1   7057  9810     9568    1762              3293          1776
2   6353  8808     7684    2405              3516          7844
3  13265  1196     4221    6404               507          1788
4  22615  5410     7198    3915              1777          5185

### Implementation: Feature Relevance
- One interesting thought to consider is if one (or more) of the six product categories is actually relevant for understanding customer purchasing. 
- That is to say, is it possible to determine whether customers purchasing some amount of one category of products will necessarily purchase some proportional amount of another category of products? 
- We can make this determination quite easily by training a supervised regression learner on a subset of the data with one feature removed, and then score how well that model can predict the removed feature.

In [11]:
from sklearn.cross_validation import train_test_split
from sklearn.tree import DecisionTreeRegressor

# TODO: Make a copy of the DataFrame, using the 'drop' function to drop the given feature
X_all = df.drop('Grocery', axis = 1)
y_all = df['Grocery']

# TODO: Split the data into training and testing sets using the given feature as the target
X_train, X_test, y_train, y_test = train_test_split(X_all, y_all, test_size=0.25, random_state=42)

# TODO: Create a decision tree regressor and fit it to the training set
clf = DecisionTreeRegressor(random_state=42)

# TODO: Report the score of the prediction using the testing set
clf.fit(X_train, y_train)
score = clf.score(X_test, y_test)

# The clf.score = r-squared
score

0.68188400854408338

**Which feature did you attempt to predict?**
- Grocery

**What was the reported prediction score?**
-  R-squared: 0.681884008544

**Detecting redundant features is a common step during feature selection so this feature necessary for identifying customers' spending habits?**
- With a coefficient of determination of .681, we know the Grocery feature does describe some of the same information found in the other features. So .319 is not enough "unique feature information" to consider it a necessary feature.

**If we repeated the regression 100x to smooth out variation in the decision tree formation and train/test splitswhat would all the prediction scores look like?**

- Fresh: -0.6662
- Milk: 0.1087
- Grocery: 0.6561
- Frozen: -1.447
- Detergents_Paper: 0.6452
- Delicatessen: -2.7253

Q1. Is it correct to say that Delicatessen is the most necissary feature because it is least described by the other categories?

Q1. I don't believe the magnitude of negative values here have any significance beyond the fact that any negative value indicates the target is not predicted well. I'm not certain though, so you might leave this question here for the next reviewer.